app/src/main/java/space/taran/arkmemo/utils/SVGUtils.kt

```diff
@@ -0,0 +1,155 @@
+package space.taran.arkmemo.utils
+
+import android.graphics.Path
+import android.util.Log
+import space.taran.arkmemo.data.viewmodels.DrawPath
+import space.taran.arkmemo.utils.SVG.Tags.viewBox
+import java.util.Stack
+
+class SVG {
+    var strokeColor = "${QUOTE}black$QUOTE"
+    private var d = ""
+
+    fun writeData(command: String) {
+        d += "$SPACE$command"
```

Comment:
We should use better data structure, like List or Dequeue.

Reply:
This is still not fixed. `SVG.pathData` uses plain string, but we should use a proper structure, at least `List`. Then, when we write it into string just use `joinToString` function.

Reply:
@ShubertMunthali still not fixed. If you used proper structures you would not need this hell:
```
commandsString.substringAfter(SPACE).substringBefore(SPACE)
```

Yes, you used `joinToString`, but `commandsString` should be structured list, too.

This doesn't look nice neither:
```
            val numberOfCommands = commandsString.filter {
                it.isLetter()
            }.length
```
With proper list, this code would be replaced by just `commands.size`.

Just never use raw strings until you really need to write something into a file or in GUI. Strings are always bad idea.